class/legacy-wallet.js

```diff
@@ -246,6 +253,10 @@ export class LegacyWallet extends AbstractWallet {
       tx.height = txs.find(t => t.tx_hash === tx.txid).height;
       tx.tx_type = findLast(txs, t => t.tx_hash === tx.txid).tx_type;
 
+      if (tx.tx_type === undefined) {
+        logger.warning('legacy-wallet', `couldn't find tx_type for new tx: ${JSON.stringify(tx)}`);
+      }
```

Comment:
Personally, I don't like the idea of having if conditions only for logging purposes. But if it'll help then I'm okay with this.